lib/Backend/arm64/ARM64UnwindEncoder.cpp

```diff
@@ -568,6 +542,69 @@ ULONG Arm64UnwindCodeGenerator::EmitFinalCodes(PBYTE buffer, ULONG bufferSize, P
     return outputIndex;
 }
 
+ULONG64 Arm64UnwindCodeGenerator::FindRegisterImmediate(int regNum, PULONG regionStart, PULONG regionEnd)
+{
+    // scan forward, looking for opcodes that assemble immediate values
+    ULONG64 pendingImmediate = 0;
+    int pendingImmediateReg = -1;
+    bool foundImmediate = false;
+    for (PULONG opcodePtr = regionStart; opcodePtr < regionEnd; opcodePtr++)
```

Comment:
a general question: shouldn't this loop run backwards from regionEnd to regionStart? if there are multiple immediates movs we will use the last one.

Reply:
It runs from loop start until the point where the register immediate is consumed. If the register is re-used in that time, only the last version is accumulated. Because of the way constants are assembled, it's a bit simpler to process them in the forward direction, since the destructive MOVZ/MOVN has to happen first.

I pulled this logic out of the main loop because it always needs to run forward. Before, in the epilog case, we were running it backwards and it was not acquiring the constants correctly.